ref/general/mp-openapi.adoc

```diff
@@ -12,19 +12,27 @@
 :page-type: general
 = Documenting microservice APIs with OpenAPI
 
-OpenAPI is a standardized mechanism to describe REST APIs. The MicroProfile OpenAPI specification is an official set of annotations and interfaces to help developers generate OpenAPI documentation for Java RESTful APIs.
+MicroProfile OpenAPI (MP OpenAPI) is a MicroProfile community programming model for OpenAPI.
+With MP OpenAPI, Java developers can generate structured documentation from their JAX-RS applications.
 
-OpenAPI, previously known as Swagger, defines a standard interface for documenting and exposing RESTful APIs. This standard interface enables both humans and computers to process a web service available through its REST API without directly accessing the source code of the web service. MicroProfile OpenAPI provides a set of Java interfaces and programming models that enable Java developers to natively produce OpenAPI documents from their JAX-RS applications.
+The structured documentation helps other developers and microservices understand and communicate with your microservice.
+Developers can write a web service to communicate with the REST API of another web service only if they have the relevant information to do so.
+MP OpenAPI specification facilitates this communication by generating human and machine readable documentation on an easy to use interface without the need to access the source code.
 
-Developers can communicate with the REST API in their web service only if they have the relevant information to do so. It is important to document APIs when they are published so that other developers can use them.
+There are two main approaches to generate an link:https://swagger.io/docs/specification/about/[OpenAPI] document, code-first and design-first.
+In the code-first approach, the developer can generate documentation of the REST API from the source code.
+The reference document then produced, lists all the API endpoints and descriptions of how to use them.
```

Comment:
This sentence reads a bit awkwardly. Maybe change "The reference document then produced lists all" to "A reference document is then produced that lists all". Also, for "descriptions of how to use them", I think it may be better to clarify what "them" is referring to in order to avoid any possible confusion.